tts_config/sample/tts/scripts/opennebula.py

```diff
@@ -178,19 +179,20 @@ def main():
             jobject = json.loads(str(base64.urlsafe_b64decode(json_data)))
 
             action = jobject['action']
+            user_info = jobject['user_info']
+            oidc = user_info['oidc']
+            iss = urlparse.urlparse(oidc['iss'])
+            iss_host = iss[1]
+            username = "%s_%s_%s" % (USER_PREFIX, iss_host, oidc['sub'])
 
             if action == "request":
-                user_info = jobject['user_info']
-                oidc = user_info['oidc']
-                iss = urlparse.urlparse(oidc['iss'])
-                iss_host = iss[1]
-                username = "%s_%s" % (iss_host, oidc['sub'])
-
                 print create_user(username, USERS_GROUP, oidc)
             elif action == "revoke":
                 state = jobject['cred_state']
-
-                print revoke_user(state)
+                if state != username:
```

Comment:
this might be bad:
If you have account linking active for a user and he linked a IAM account with a different OpenID Connect account e.g. Google:
1st logs in with IAM and creates user
2nd logs out
3rd logs in with google
4th revokes
in this case the opennebula script would return "bad state" error although from a user point of view everything is okay.